src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ILGeneratorImpl.cs

```diff
@@ -86,98 +133,91 @@ public override void Emit(OpCode opcode, int arg)
             {
                 if (arg >= -1 && arg <= 8)
                 {
-                    _il.OpCode(arg switch
+                    EmitOpcode(arg switch
                     {
-                        -1 => ILOpCode.Ldc_i4_m1,
-                        0 => ILOpCode.Ldc_i4_0,
-                        1 => ILOpCode.Ldc_i4_1,
-                        2 => ILOpCode.Ldc_i4_2,
-                        3 => ILOpCode.Ldc_i4_3,
-                        4 => ILOpCode.Ldc_i4_4,
-                        5 => ILOpCode.Ldc_i4_5,
-                        6 => ILOpCode.Ldc_i4_6,
-                        7 => ILOpCode.Ldc_i4_7,
-                        _ => ILOpCode.Ldc_i4_8,
+                        -1 => OpCodes.Ldc_I4_M1,
+                        0 => OpCodes.Ldc_I4_0,
+                        1 => OpCodes.Ldc_I4_1,
+                        2 => OpCodes.Ldc_I4_2,
+                        3 => OpCodes.Ldc_I4_3,
+                        4 => OpCodes.Ldc_I4_4,
+                        5 => OpCodes.Ldc_I4_5,
+                        6 => OpCodes.Ldc_I4_6,
+                        7 => OpCodes.Ldc_I4_7,
+                        _ => OpCodes.Ldc_I4_8
                     });
                     return;
                 }
 
                 if (arg >= -128 && arg <= 127)
                 {
-                    _il.OpCode(ILOpCode.Ldc_i4_s);
-                    _builder.WriteSByte((sbyte)arg) ;
+                    Emit(OpCodes.Ldc_I4_S, (sbyte)arg);
                     return;
                 }
             }
             else if (opcode.Equals(OpCodes.Ldarg))
             {
                 if ((uint)arg <= 3)
                 {
-                    _il.OpCode(arg switch
+                    EmitOpcode(arg switch
                     {
-                        0 => ILOpCode.Ldarg_0,
-                        1 => ILOpCode.Ldarg_1,
-                        2 => ILOpCode.Ldarg_2,
-                        _ => ILOpCode.Ldarg_3,
+                        0 => OpCodes.Ldarg_0,
+                        1 => OpCodes.Ldarg_1,
+                        2 => OpCodes.Ldarg_2,
+                        _ => OpCodes.Ldarg_3,
```

Comment:
This condition will handle all negative numbers too, is that what we want? I think an exception or some other other mode would be more appropriate.

Reply:
I thought when casting to `uint` negative values would become positive large number something that > 3, no?

Reply:
Yes, that is true. My statement was more related to the notion of all unknown values being converted to `Ldarg_3`. I think `Ldarg_0` or an exception being the failure mode makes more sense.

Reply:
I am OK with defaulting to `Ldarg_0`, though this is how the existing CoreLib `ILGenerator` does, maybe we should keep it as is for compat, what you think?

Reply:
Boo. Your instinct here is correct - keep the same behavior.